src/manifest_parser.cc

```diff
@@ -141,31 +141,37 @@ bool ManifestParser::ParseRule(string* err) {
   if (env_->LookupRuleCurrentScope(name) != NULL)
     return lexer_.Error("duplicate rule '" + name + "'", err);
 
-  Rule* rule = new Rule(name);  // XXX scoped_ptr
+  Rule* rule = new Rule(name);
```

Comment:
use raii

```cpp
#ifdef cpp11 
    struct Unleaker
    {
        Unleaker(Rule* rule) : m_rule(rule){}
        ~Unleaker(){ delete m_rule; }
        Rule* get(){ return m_rule; }
        Rule*m_rule;
    };
#else
    using Unleaker = std::unique_ptr<Rule>;
#endif
    Unleaker rule(new Rule(name));
```
    then reference as "rule.get()" instead of rule.